src/linearshrinkage.jl

```diff
@@ -315,6 +318,7 @@ function linear_shrinkage(::DiagonalCommonVariance, Xc::AbstractMatrix,
                           corrected::Bool, weights::FrequencyWeights...)
 
     F   = target_B(S, p)
+    p -= sum(iszero, diag(S))
```

Comment:
I'm not sure if doing this by default is the right thing. If p > n we always have zero-variance directions we don't adjust for, so subtracting them if they happen to align with one of the components seems a bit arbitrary. I'd suggest adding a parameter controlling whether we do it or not (and by default have the old behavior to not make it a breaking change).

Reply:
I'm not sure those are entirely comparable, as the diagonal is rather special for many of these shrinkage methods. Still, the idea of making this configurable makes sense.

Reply:
Yes, they are not entirely comparable but I can see good arguments for both behaviors in different situations so making it configurable is the best option.